iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -234,52 +237,62 @@ def _adjust_operator_bundle(manifests_path, metadata_path, organization=None, pi
         elif customization_type == 'registry_replacements':
             registry_replacements = customization.get('replacements', {})
             log.info('Resolving image pull specs and applying registry replacements')
-            bundle_metadata = _get_bundle_metadata(operator_manifest, pinned_by_iib)
+            bundle_metadata = _get_bundle_metadata(
+                operator_manifest, pinned_by_iib, pinning_metadata=True
+            )
             _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_replacements)
         elif customization_type == 'csv_annotations' and organization:
             org_csv_annotations = customization.get('annotations')
             if org_csv_annotations:
                 log.info('Applying csv annotations for organization %s', organization)
                 _adjust_csv_annotations(operator_manifest.files, package_name, org_csv_annotations)
+        elif customization_type == 'image_name_from_labels':
+            org_image_name_template = customization.get('template', '')
+            if org_image_name_template:
+                new_bundle_metadata = _get_bundle_metadata(operator_manifest, pinned_by_iib)
+                _replace_image_name_from_labels(new_bundle_metadata, org_image_name_template)
 
     return labels
 
 
-def _get_bundle_metadata(operator_manifest, pinned_by_iib):
+def _get_bundle_metadata(operator_manifest, pinned_by_iib, pinning_metadata=False):
     """
     Get bundle metadata i.e. CSV's and all relatedImages pull specifications.
 
     :param operator_manifest.operator.OperatorManifest operator_manifest: the operator manifest
         object.
     :param bool pinned_by_iib: whether or not the bundle image has already been processed by
         IIB to perform image pinning of related images.
+    :param bool pinning_metadata: boolean specifying if metadata should be filtered for pinning
```

Comment:
After reading the code, it makes sense what the purpose of this parameter is. Can we consider renaming it something like `ignore_sanity_checks` ? I think that better describes its intent.

```
:param bool ignore_sanity_check: boolean specifying if sanity checks should be skipped.
```

I don't think that's great either, but maybe slightly better?